infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java

```diff
@@ -445,7 +445,8 @@ public void markSyncStatus(SyncStatus status) {
     public void startSyncStatusUpdatePeriodicTask() {
         log.info("Start sync status update periodic task");
         lastAckedTsPoller = Executors.newSingleThreadScheduledExecutor(
-                new ThreadFactoryBuilder().setNameFormat("ack-timestamp-reader").build());
+            new ThreadFactoryBuilder().setNameFormat(
+                "ack-timestamp-reader-"+remoteClusterId).build());
```

Comment:
I am not aware of there being a limit on the length of a log line but can confirm.

I do agree on the thread name being long though.  We will have to think of a shorter version once the session is introduced as it will have more info.  Maybe a shortened hash of the name?

Reply:
I was thinking of the same for my connection PR...I will need to embed the session info somehow to differentiate the clients of a channel.

One thought around the hash I had was that, to make it short, we will have to truncate it. And then that could have collisions. Still thinking how we can achieve it..